<?xml version="1.0" encoding="UTF-8"?>
<launch>
<node pkg="rviz" type="rviz" name="rviz"/>
<node pkg="rosbag" type="play" name="rosbag_play" output="screen" required="true"
         args="/home/nehalnevle/internship_int2peta/image_pose_ws/src/yolov5_pytorch_ros/bag/2nd.bag"/>
    <!-- Camera topic and weights, config and classes files -->
    <arg name="image_topic"	                default="/camera1/image_raw"/>
    <arg name="weights_path"	            default="$(find yolov5_pytorch_ros)/weights/best.pt"/>

    <!-- Published topics -->
    <arg name="publish_image"	            default="true"/>
    <arg name="detected_objects_topic"      default="detected_objects_in_image"/>
    <arg name="detections_image_topic"      default="detections_image_topic"/>

    <!-- Detection confidence -->
    <arg name="confidence"                  default="0.10"/>

    <!-- Node -->
    <node name="detector" pkg="yolov5_pytorch_ros" type="detect.py" output="screen" respawn="true">
        <param name="image_topic"               value="$(arg image_topic)"/>
        <param name="frame_id"               		value="camera"/>
        <param name="weights_path"              value="$(arg weights_path)"/>
        <param name="publish_image"             value="$(arg publish_image)"/>
        <param name="detected_objects_topic"    value="$(arg detected_objects_topic)"/>
        <param name="detections_image_topic"    value="$(arg detections_image_topic)"/>
        <param name="confidence"                value="$(arg confidence)"/>
    </node>
</launch>
